fuse/ipns/ipns_unix.go

```diff
@@ -118,14 +118,7 @@ func loadRoot(ctx context.Context, rt *keyRoot, ipfs *core.IpfsNode, name string
 
 	rt.root = root
 
-	switch val := root.GetValue().(type) {
```

Comment:
So, I need to get better about submitting comments instead of editing them till my browser crashes and they disappear...

This is the one real reason to allow the root to be something other than a directory. However, really, the "root" is just a way to tie a file/directory in an MFS tree to a publisher. I'd rather just:

1. Hold a directory directly as you suggested here https://github.com/ipfs/go-ipfs/issues/5066#issuecomment-401616027.
2. Provide some way to "watch" files/directories.

Currently, we have the `childCloser` but I wonder if we want a more general system. Thoughts?

Reply:
> So, I need to get better about submitting comments instead of editing them till my browser crashes and they disappear...

I'm glad I'm not the only one, I'm seriously considering submitting a feature request to GitHub that would allow me to explicitly save my drafts..

> Thoughts?

I was considering adding that logic to the directory implementation of `childCloser`, if the directory has a parent call the parent's `closeChild` (current behavior), if not (you are the root) call the republisher and tell it things have changed. That would require every directory to be aware of the republisher, either with a pointer to it or some other way (which is not nice). Anyway, I need to keep studying how the republisher works and see if I can come up with something better.

Reply:
I was thinking of a more general purpose notification boadcast system where users could register and unregister any number of notifiees (where the parent directory would register itself as a notifiee).

Reply:
Could you elaborate a bit more on that? I thought the directory was the notifier (I may be misunderstanding the word *notifiee*, that's the one that receives the notification, right?).

Reply:
Unrelated:
You can save a code comment with a "Start Review", comments will be visible only to you until you send the review.